File: form-sync/main/src/lib/schema.ts
export const EXCEL_PRIMARY_ID_NAME = '__StarionId';
export const EXCEL_HASHED_PRIMARY_ID = 'f_gfbbfabeggejigfgbfhdcdbecifcjhdd';
export const GOOGLE_SHEETS_PRIMARY_ID_NAME = '__StarionId';
export const GOOGLE_SHEETS_HASHED_PRIMARY_ID =
  'f_gfbbfabeggejigfgbfhdcdbecifcjhdd';

export type RecordId = string;
export type FieldId = string;

export enum NativeDataType {
  String = 'String',
  Number = 'Number',
  Boolean = 'Boolean',
  Date = 'Date',
  List = 'List',
  Object = 'Object',
}

export type SchemaField = {
  id: string;
  name: string;
  type: NativeDataType;
  originalType: string;
  nullable: boolean;
  enum: any[];
  readonly: boolean;
  primary: boolean;
};

export type DataSourceSchema = Record<FieldId, SchemaField>;

File: apps/processors/downloader/libs/schema/schema.go
package schema

const HashedPrimaryField = "f_gfbbfabeggejigfgbfhdcdbecifcjhdd" // MD5 hash of OriginalPrimaryFieldName
const OriginalPrimaryFieldName = "__StarionId"
const PrimaryFieldName = "id"

type DataType string

const (
	String  DataType = "String"
	Number  DataType = "Number"
	Date    DataType = "Date"
	Boolean DataType = "Boolean"
	Array   DataType = "Array"
	Unknown DataType = "Unknown"
)

type TableSchema map[string]FieldSchema

type FieldSchema struct {
	Name         string        `json:"name"`
	Type         DataType      `json:"type"`
	OriginalType string        `json:"originalType"`
	Nullable     bool          `json:"nullable"`
	Enum         []interface{} `json:"enum"`
	Readonly     bool          `json:"readonly"`
	Primary      bool          `json:"primary"`
	// Format       string        `json:"format"`
}
